packages/cli/src/registry.ts

```diff
@@ -174,18 +181,24 @@ export async function createDefaultReadRegistry(
 
   const localRegistry = new LocalRegistry(settings.cannonDirectory);
   const onChainRegistry = new OnChainRegistry({ signerOrProvider: provider, address: settings.registryAddress });
+  const inMemoryRegistry = new InMemoryRegistry();
 
   if (!(await isConnectedToInternet())) {
     debug('not connected to internet, using local registry only');
     // When not connected to the internet, we don't want to check the on-chain registry version to not throw an error
     console.log(yellowBright('⚠️  You are not connected to the internet. Using local registry only'));
-    return new FallbackRegistry([...additionalRegistries, localRegistry]);
+    return new FallbackRegistry([...additionalRegistries, inMemoryRegistry, localRegistry]);
   } else if (settings.registryPriority === 'local') {
     debug('local registry is the priority, using local registry first');
-    return new FallbackRegistry([...additionalRegistries, localRegistry, onChainRegistry]);
+    return new FallbackRegistry([...additionalRegistries, inMemoryRegistry, localRegistry, onChainRegistry]);
   } else {
     debug('on-chain registry is the priority, using on-chain registry first');
-    const fallbackRegistry = new FallbackRegistry([...additionalRegistries, onChainRegistry, localRegistry]);
+    const fallbackRegistry = new FallbackRegistry([
+      ...additionalRegistries,
+      inMemoryRegistry,
```

Comment:
in memory registry should always be first in the array, not after additionals.

Reply:
also, please *double check* the `FallbackRegistry` as it is now, I thik it has a `InMemoryRegistry` built into it now, so its unnecessary (and harmful!) to add the in memory registry here.

Reply:
FallbackRegistry did have the InMemoryRegistry built into it, so I removed the changes I added.